app/src/main/java/com/fstyle/androidtrainning/screen/main/playingfragments/lyrics/LyricsFragment.java

```diff
@@ -27,11 +56,76 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         View v = inflater.inflate(R.layout.fragment_lyrics, container, false);
         initViews(v);
 
+        ((MainActivity)getActivity()).setOnTrackChangeListener(this);
+
+        try {
+            File sdCard = new File(Constant.STR_PATH_UPPER_4);
+            fileArrayList = LyricSdCard.findLyrics(sdCard);
+        } catch (Exception e) {
+            File sdCard = new File(Constant.STR_PATH_LOWER_4);
+            fileArrayList = LyricSdCard.findLyrics(sdCard);
+        }
+        
+        initReceiver();
         return v;
     }
 
+    private void initReceiver() {
+        mReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                switch (intent.getAction()) {
+                    case Constant.ACTION_MEDIA_TIME:
+                        mLyricView.setCurrentTimeMillis(
+                                intent.getIntExtra(Constant.EXTRA_CURRENT_TIME, 0));
+                        break;
+                    default:
+                        break;
+                }
+            }
+        };
+        getActivity().registerReceiver(mReceiver, currentTimeFilter);
+    }
+
     private void initViews(View v) {
         mPresenter = new LyricsPresenter();
         mPresenter.setView(this);
+        mLyricView = v.findViewById(R.id.lyricView);
+    }
+
+    private void loadLyric(Track track) {
+
+        String songPath = track.getTrackData();
+        int indexSongData = songPath.lastIndexOf("/");
```

Comment:
WARNING: [UseIndexOfChar] String.indexOf(char) is faster than String.indexOf(String).
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#UseIndexOfChar